tests/XrdHttpTests/XrdHttpTests.cc

```diff
@@ -574,3 +574,78 @@ TEST(XrdHttpTests, xrdHttpReadRangeHandlerMultiChunksTwoRanges) {
     ASSERT_EQ(false, static_cast<bool>(error));
   }
 }
+
+TEST(XrdHttpTests, strDecodeEncodeTest) {
+  std::string input = "zteos64%3AMDAF5PGJ4Wa12g%3D";
+  {
+    std::string decoded(decode_str(input.c_str()));
+    std::string encoded(encode_str(decoded.c_str()));
+    ASSERT_EQ(input, encoded);
```

Comment:
How about having the long list of encoded/decoded strings, and using a simple loop over all cases in the tests?

```cpp
const std::pair<std::string, std::string> input_pairs = { ... };

for(auto [encoded, decoded] : input_pairs) {
    ASSERT_EQ(decode_str(encoded), decoded);
    ASSERT_EQ(encoded, encode_str(decoded));
    ASSERT_EQ(encode_str(decode_str(encoded)), encoded);
    ASSERT_EQ(decode_str(encode_str(decoded)), decoded);
}
```

Reply:
Yes I did it but did not push yet ;)